src/gobapi/graphql/schema.py

```diff
@@ -4,30 +4,103 @@
 
 """
 import graphene
+import re
 
-from graphene_sqlalchemy import SQLAlchemyObjectType, SQLAlchemyConnectionField
+from graphene_sqlalchemy import SQLAlchemyObjectType
 
 from gobcore.model import GOBModel
-from gobcore.model.metadata import PRIVATE_META_FIELDS, PUBLIC_META_FIELDS, FIXED_FIELDS
 from gobcore.model.sa.gob import models
 
-exclude_fields = tuple(name for name in {
-    **PRIVATE_META_FIELDS,
-    **PUBLIC_META_FIELDS,
-    **FIXED_FIELDS
-}.keys())
-queries = []
-model = GOBModel()
-for catalog_name, catalog in model.get_catalogs().items():
-    for collection_name, collection in model.get_collections(catalog_name).items():
+from gobapi.graphql import graphene_type, exclude_fields
+from gobapi.graphql.filters import FilterConnectionField, get_resolve_attribute
+
+
+def _get_sorted_references(model):
+    """Get an ordered list of references
+
+    A reference is a "catalogue:collection" string
+
+    Each collection in the given model might refer to another collection.
+
+    For each collection in the model, all references to other collections are collected
+    these references are stored in a dictionary
+
+    Then a list is constructed so that when A refers to B, B will be before A in the list
+
+    :param model: the model that contains all catalogs and collections
+    :return: a sorted list of references ("catalogue:collection")
+    """
+    refs = {}
+    for catalog_name, catalog in model.get_catalogs().items():
+        for collection_name, collection in model.get_collections(catalog_name).items():
+            from_ref = f"{catalog_name}:{collection_name}"
+
+            # Get all references fields for the collection
+            ref_types = ["GOB.Reference"]
+            # Get all references that are contained in these fields
+            refs[from_ref] = [ref["ref"] for ref in collection["attributes"].values() if ref["type"] in ref_types]
```

Comment:
In gobcore.model all references are already being added to the model in collection['references']. This might save some duplication in this project and core. 

Maybe extend that code in core with the sorted code below, so we can use that in other repo's as well.